lib/types/src/store_id.rs

```diff
@@ -0,0 +1,22 @@
+use std::{
+    num::NonZeroU64,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+/// Unique ID to identify a context.
+///
+/// Every handle to an object managed by a context also contains the ID of the
+/// context. This is used to check that a handle is always used with the
+/// correct context.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct StoreId(NonZeroU64);
+
+impl Default for StoreId {
+    // Allocates a unique ID for a new context.
+    fn default() -> Self {
+        // No overflow checking is needed here: overflowing this would take
+        // thousands of years.
+        static NEXT_ID: AtomicU64 = AtomicU64::new(1);
```

Comment:
Maybe we should use `AtomicU64` or `AtomicUsize` consistantly, even if this end up beeing the same things on our platforms

Reply:
Fixed